pkg/envoy/lds/response.go

```diff
@@ -27,47 +30,88 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
 		statsHeaders = proxy.StatsHeaders()
 	}
 
-	lb := newListenerBuilder(meshCatalog, proxy.Identity, cfg, statsHeaders, cm.GetTrustDomain())
+	svcList, err := proxyRegistry.ListProxyServices(proxy)
+	if err != nil {
+		log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrFetchingServiceList)).
+			Str("proxy", proxy.String()).Msgf("Error looking up MeshServices associated with proxy")
+		return nil, err
+	}
 
 	// --- OUTBOUND -------------------
-	outboundListener, err := lb.newOutboundListener()
-	if err != nil {
-		log.Error().Err(err).Str("proxy", proxy.String()).Msg("Error building outbound listener")
+	outboundLis := ListenerBuilder().
+		Name(OutboundListenerName).
+		ProxyIdentity(proxy.Identity).
+		Address(constants.WildcardIPAddr, constants.EnvoyOutboundListenerPort).
+		TrafficDirection(xds_core.TrafficDirection_OUTBOUND).
+		DefaultOutboundListenerFilters().
+		PermissiveMesh(cfg.IsPermissiveTrafficPolicyMode()).
+		PermissiveEgress(cfg.IsEgressEnabled()).
+		OutboundMeshTrafficPolicy(meshCatalog.GetOutboundMeshTrafficPolicy(proxy.Identity)).
+		ActiveHealthCheck(cfg.GetFeatureFlags().EnableEnvoyActiveHealthChecks)
+
+	if cfg.GetMeshConfig().Spec.Traffic.EnableEgress {
+		outboundLis.PermissiveEgress(true)
 	} else {
-		if outboundListener == nil {
-			// This check is important to prevent attempting to configure a listener without a filter chain which
-			// otherwise results in an error.
-			log.Debug().Str("proxy", proxy.String()).Msg("Not programming nil outbound listener")
-		} else {
-			ldsResources = append(ldsResources, outboundListener)
+		egressPolicy, err := meshCatalog.GetEgressTrafficPolicy(proxy.Identity)
+		if err != nil {
+			return nil, fmt.Errorf("error building LDS response: %w", err)
 		}
+		outboundLis.EgressTrafficPolicy(egressPolicy)
+	}
+	if cfg.GetMeshConfig().Spec.Observability.Tracing.Enable {
+		outboundLis.TracingEndpoint(cfg.GetTracingEndpoint())
+	}
+	if cfg.GetMeshConfig().Spec.FeatureFlags.EnableWASMStats {
```

Comment:
Would this be a good opportunity to remove other knobs from featureflags, or do we want to keep it here?

Reply:
This change is definitely not relevant to making API changes. Feel free to open an issue to discuss this further.